medvis/dicom/src/io/gdcmvolumereader.cpp

```diff
@@ -0,0 +1,1070 @@
+                                        // dont read whole image (gdcm::ImageRegionReader)
+        std::ifstream imageInputStream(imgInfo.path, std::ios::binary);
+
+        if (!imageInputStream.is_open()) {
+            // could not open file => skip
+            continue;
+        }
+
+        imageReader.SetStream(imageInputStream);
+        if (!imageReader.CanRead()) {
+            // not a DICOM file => skip
+            continue;
+        }
+
+        if (imageReader.Read()) {
+            gdcm::Image image = imageReader.GetImage();
+            imageStack.push_back(image);
+        }
+    }
+
+    // Metadata from images
+
+    size_t channelCount = 0;     // e.g. 3 for RGB
+    size_t channelBits = 0;      // bit width of each channel
+    size_t usedChannelBits = 0;  // bit width of data in each channel
+    unsigned short isSignedInt = 0;
+    gdcm::PixelFormat pixelformat;
+
+    // Origin can be used to transform the xy image grid into xyz world space
+    glm::dvec3 origin;
+
+    // Spacing between voxels in millimeters
+    // note that this must be reconstructed by GDCM with varying accuracy
+    // http://gdcm.sourceforge.net/wiki/index.php/Imager_Pixel_Spacing#ImagerPixelSpacing_to_PixelSpacing
+    glm::dvec3 spacing;
+
+    // Rescale describes the conversion from disk data to hounsfield value
+    // https://blog.kitware.com/dicom-rescale-intercept-rescale-slope-and-itk/
+    double rescaleIntercept, rescaleSlope;
+
+    // Photometric interpretation string
+    // e.g. MONOCHROME1 where lowest value is white or MONOCHROME2 where lowest value is black
+    // https://dicom.innolitics.com/ciods/mr-image/image-pixel/00280004
+    std::string piString;
+
+    if (imageStack.size() > 0) {
+        const gdcm::Image& image =
+            imageStack[0];  // just assume that all images have consistent values
+
+        auto colorspace = image.GetPhotometricInterpretation();
+        colorspace.GetString();
+        channelCount = colorspace.GetSamplesPerPixel();
+
+        origin.x = image.GetOrigin(0);
+        origin.y = image.GetOrigin(1);
+        origin.z = image.GetOrigin(2);
+
+        spacing.x = image.GetSpacing(0);
+        spacing.y = image.GetSpacing(1);
+
+        if (sliceThickness == std::numeric_limits<double>::infinity()) {
+            spacing.z = image.GetSpacing(2);  // fallback if no explicit slice thickness is set
+        } else {
+            spacing.z =
+                sliceThickness;  // image.GetSpacing(2); seems to be not set correctly for volumes?
+        }
+
+        pixelformat = image.GetPixelFormat();
+
+        rescaleIntercept = image.GetIntercept();
+        rescaleSlope = image.GetSlope();
+
+        channelBits = image.GetPixelFormat().GetBitsAllocated();
+        usedChannelBits = image.GetPixelFormat().GetBitsStored();
+        isSignedInt = image.GetPixelFormat().GetPixelRepresentation();
+    } else {
+        return 0;  // empty volume
```

Comment:
Should throw, not return 0

Reply:
exists no longer due to refactoring of `GdcmVolumeReader::getVolumeDescription()`